core/modules/layout/tests/layout.test

```diff
@@ -1365,6 +1262,244 @@ class LayoutInterfaceTest extends BackdropWebTestCase {
+    $this->clickLink(t('Add block'), 2);
+    $this->clickLink(t('Add a hero block'));
+
+    // Confirm a too-small background image fails validation.
+    $edit = array(
+      'files[image]' => $bad_realpath_filename,
+    );
+    $this->backdropPost(NULL, $edit, t('Add block'));
+    $this->assertText(t('The specified file ' . $bad_filename . ' could not be uploaded. The image is too small; the minimum dimensions are 1200x300 pixels.'));
+
+    // Resubmit using the good image.
+    $edit['files[image]'] = $good_realpath_filename2;
+    $this->backdropPost(NULL, $edit, t('Add block'));
+
+    // Save the layout.
+    $this->backdropPost(NULL, array(), t('Save layout'));
+
+    // Go to the user page and check for all hero blocks.
+    $this->backdropGet('user');
+
+    // Get all hero blocks on the user page and confirm the correct number.
+    $elements = $this->xpath('//*[contains(@class,:region)]//*[contains(@class,:block)]', array(
+      ':region' => 'l-content',
+      ':block' => 'block-layout-hero-block',
+    ));
+    $this->assertEqual(count($elements), 3, 'Three hero blocks were found in the content area.');
+
+    // Get all the hero block elements for testing. Order matters here.
+    $elements = $this->xpath('(//div[contains(@class, "block-layout-hero-block")])');
+
+    // Block 1: Confirm both title and content are visible.
+    $this->assertText($hero_title_1);
+    $this->assertText($hero_content_1);
+    // Block 1: Confirm handy `block-hero-no-image` class is present.
+    $classes_1 = (string)$elements[0]['class'];
+    $this->assertTrue(strstr($classes_1, 'block-hero-no-image'));
+    // Block 1: Confirm that no background image is present.
+    $this->assertNull($elements[0]['style'], 'No image present for 1st hero block.');
+
+    // Block 2: no title present.
+    $title_2 = $elements[1]->xpath('.//h2');
+    $this->assertFalse($title_2, 'No title present for 2nd hero block.');
+    // Block 2: content is visible.
+    $this->assertText($hero_content_2);
+    // Block 2: Confirm handy `block-hero-image` class is present.
+    $classes_2 = (string)$elements[1]['class'];
+    $this->assertTrue(strstr($classes_2, 'block-hero-image'));
+    // Block 2: Background image present.
+    $style_2 = (string)$elements[1]['style'];
+    $this->assertTrue(strstr($style_2, $good_filename), 'Image present for 2nd hero block.');
+
+    // Block 3: no title present.
+    $title_3 = $elements[2]->xpath('.//h2');
+    $this->assertFalse($title_3, 'No title present for 3rd hero block.');
+    // Block 3: no content present.
+    $content_3_element = $elements[2]->xpath('.//div[contains(@class, "block-content")]');
+    // Extract the content.
+    $content_3 = (string)$content_3_element[0];
+    // Remove non-breaking spaces.
+    $content_3 = preg_replace('~\x{00a0}~siu', '', $content_3);
+    // Trim all other whitespace.
+    $content_3 = trim(preg_replace('/\s+/S', '', $content_3), ' \t\n\r\0\x0B\xC2\xA0');
```

Comment:
Yikes, I finally found a way to represent this a (little) easier. The problem is SimpleXML is returning a unicode character, but PHP doesn't support unicode strings natively until 7.0.0, so I came up with this:

```php
    // We inserted an HTML non-breaking space, but SimpleXML will return a
    // unicode non-breaking space (\u00a0). PHP <7.0.0 does not have native
    // unicode strings, so we encode as JSON which will encode the character.
    // In PHP 7 this could be represented simply as $content_3 === "\u00a0".
    $unicode_nbsp = json_encode(trim($content_3), '"');
    $this->assertEqual($unicode_nbsp, '\u00a0', '3rd hero block contains only a non-breaking space.');
```